hbook/s2d.go

```diff
@@ -35,11 +35,31 @@ func NewS2D(pts ...Point2D) *S2D {
 // NewS2DFrom creates a new 2-dim scatter with x,y data slices.
 //
 // It panics if the lengths of the 2 slices don't match.
-func NewS2DFrom(x, y []float64) *S2D {
+// Two first options are used to get eventual errors (with panic if sizes don't match)
+// Supports only symmetric errors
+func NewS2DFrom(x, y []float64, opts... []float64) *S2D {
```

Comment:
Sure, I wasn't sure what was the spirit of Go regarding code duplication, but I'll write down explicitly the 3 functions you mentionned.